packages/react-query/src/useQuery.ts

```diff
@@ -59,6 +59,34 @@ export function useQuery<
   > & { initialData: TQueryFnData | (() => TQueryFnData) },
 ): DefinedUseQueryResult<TData, TError>
 
+export function useQuery<
+  TQueryFnData = unknown,
+  TError = unknown,
+  TData = TQueryFnData,
+  TQueryKey extends QueryKey = QueryKey,
+>(
+  queryKey: TQueryKey,
+  queryFn: QueryFunction<TQueryFnData, TQueryKey>,
+  options?: Omit<
+    UseQueryOptions<TQueryFnData, TError, TData, TQueryKey>,
+    'queryKey' | 'queryFn'
+  >,
+): UseQueryResult<TData, TError>
+
+export function useQuery<
+  TQueryFnData = unknown,
+  TError = unknown,
+  TData = TQueryFnData,
+  TQueryKey extends QueryKey = QueryKey,
+>(
+  queryKey: TQueryKey,
+  queryFn: QueryFunction<TQueryFnData, TQueryKey>,
+  options?: Omit<
+    UseQueryOptions<TQueryFnData, TError, TData, TQueryKey>,
+    'queryKey' | 'queryFn'
+  >,
+): DefinedUseQueryResult<TData, TError>
```

Comment:
this would need to be `UseQueryResult`:

```suggestion
): UseQueryResult<TData, TError>
```

Reply:
I added 2 overloads to follow your pattern, one that uses `UseQueryResult` and the other one uses `DefinedUseQueryResult`. Should I just leave the one with `UseQueryResult`?

Reply:
the result is `Defined` if we have an `initialData` other than undefined, or a function that returns something that is not undefined.

so I think we would need one last overload for each "version" (the one with 3, 2 and one parameter) that returns a normal `UseQueryResult` - meaning 3 new overloads in total.

Reply:
Cool, I just added them 😄. Could you please check?

Reply:
looks right 🚀